packages/subapp-server/lib/fastify-plugin.js

```diff
@@ -41,13 +43,20 @@ function makeRouteHandler({ path, routeRenderer, routeOptions }) {
 
       const data = context.result;
       const status = data.status;
+      
+      const unsafeEval = process.env.NODE_ENV !== "production" ? 
+        `"unsafe-eval" "unsafe-inline"` : "";
+
+      const styleSrc = styleNonce ? `style-src 'nonce-${styleNonce}' 'strict-dynamic';` : "";
+      const scriptSrc = scriptNonce ? `script-src 'nonce-${scriptNonce}' 'strict-dynamic' ${unsafeEval};`: "";
 
-      const nonce = setCSPHeader({ routeOptions });
-      const unsafeEval = process.env.NODE_ENV !== "production" ? `'unsafe-eval'` : "";
-      if (nonce) {
-        reply.header("Content-Security-Policy", 
-        `script-src 'nonce-${nonce}' 'strict-dynamic' ${unsafeEval};`);
+      if (scriptSrc) {
+        reply.header("Content-Security-Policy", scriptSrc);
       }
+      if (styleSrc) {
+        reply.header("Content-Security-Policy", styleSrc);
```

Comment:
Wondering if this will override the previously set header value. Would it be possible to join scriptSrc and styleSrc and set that to header?